### PR TITLE
feature/covariance interpolator

### PR DESCRIPTION
**Pull request by rhayes777**
First cut of an interpolator that respects covariance. The interpolator must be made using a list of PDFSamples such that covariance information is available.

```python
interpolator = CovarianceInterpolator(pdf_samples)
instance = interpolator(interpolator.t == 1)
print(instance.gaussian.centre)
```

DynestyStatic is used and the relationship between each variable and time is assumed to be linear.

Seems to give correct results but more testing is required.

### CODE REVIEW

**Comment by Jammy2211** [comment]
> The interpolator must be made using a list of PDFSamples such that covariance information is available.

Does this use more information than the covariance matrix?

**Comment by rhayes777** [comment]
No just the covariance matrix

**Comment by Jammy2211** [comment]
Ok we'll be putting that out as a separate .csv file soon enough!

File: autofit/interpolator/covariance.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def __repr__(self):
         return str(self)
 
 
-class LinearAnalysis(Analysis):
+class CovarianceAnalysis(Analysis):
```
Comment (Jammy2211) [inline autofit/interpolator/covariance.py:43]: We're using autofit in autofit!

**Review by Jammy2211 — APPROVED** [review]

**Comment by Jammy2211** [comment]
I will set up a CTI example when I have time and add visualization.

File: autofit/interpolator/covariance.py (hunk 1 above)
Comment (rhayes777) [reply autofit/interpolator/covariance.py:43]: Autoception

**Pull request merged**